streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicProperties.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+public class InternalTopicProperties {
```

Comment:
Thought about it, but it felt more natural to introduce separate class rather than using `RepartitionTopicConfig` in current design. `RepartitionTopicConfig` potentially can be package-private (haven't touched this in current PR) and I wanted to avoid leaking it through other packages. It felt like best to leave construction of `RepartitionTopicConfig` in `InternalTopologyBuilder` class since `InternalTopologyBuilder` "knows the best" how to construct it. Idea of this class is to provide bare minimum configs of internal topic properties. Hope my way of thinking around this makes sense. wdyt?